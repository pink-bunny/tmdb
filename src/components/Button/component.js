import React from 'react';
import PropTypes from 'prop-types';
import { Button as AntdButton } from 'antd';

const Button = ({
  text,
  type,
  htmlType,
  loading
}) => (
  <AntdButton
    loading={loading}
    type={type}
    htmlType={htmlType}
  >
    {text}
  </AntdButton>
);

Button.propTypes = {
  text: PropTypes.string,
  type: PropTypes.string,
  htmlType: PropTypes.string,
  loading: PropTypes.bool
};
Button.defaultProps = {
  text: 'Button',
  type: 'primary',
  htmlType: 'button',
  loading: false
};

export default Button;
