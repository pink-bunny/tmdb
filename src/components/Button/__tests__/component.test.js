import React from 'react';
import { shallow } from 'enzyme';

import Button from '../index';

describe('Button component', () => {
  it('matches snapshot without props', () => {
    const wrapper = shallow(<Button />);
    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot with custom props', () => {
    const props = {
      text: 'Custom button',
      type: 'dashed',
      htmlType: 'submit',
      loading: true,
      onClick: jest.fn()
    };
    const wrapper = shallow(<Button {...props} />); /* eslint-disable-line */

    expect(wrapper).toMatchSnapshot();
  });
});
