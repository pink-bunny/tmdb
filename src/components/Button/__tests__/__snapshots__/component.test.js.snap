// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component matches snapshot with custom props 1`] = `
<Button
  block={false}
  ghost={false}
  htmlType="submit"
  loading={true}
  onClick={[MockFunction]}
  type="dashed"
>
  Custom button
</Button>
`;

exports[`Button component matches snapshot without props 1`] = `
<Button
  block={false}
  ghost={false}
  htmlType="button"
  loading={false}
  type="primary"
>
  Button
</Button>
`;
